src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.cs

```diff
@@ -162,12 +177,16 @@ private bool StartNewModelComputation(ICompletionService completionService, Comp
 
             this.sessionOpt = new Session(this, computation, GetCompletionRules(), Presenter.CreateSession(TextView, SubjectBuffer, null));
 
-            var completionProviders = triggerInfo.TriggerReason == CompletionTriggerReason.Snippets
+            var completionProviders = trigger is DisplaySnippetsCompletionTrigger
```

Comment:
We could. In the case of some of the triggers, there's data added as well (e.g. DeletedCharacter).